aiida/storage/psql_dos/alembic_cli.py

```diff
@@ -38,8 +39,8 @@ def execute_alembic_command(self, command_name, connect=True, **kwargs):
             raise click.ClickException('No profile specified')
         migrator = PsqlDosMigrator(self.profile)
 
-        context = migrator._alembic_connect() if connect else nullcontext(migrator._alembic_config())  # pylint: disable=protected-access
-        with context as config:  # type: ignore[attr-defined]
+        context = migrator._alembic_connect() if connect else contextlib.nullcontext(migrator._alembic_config())  # pylint: disable=protected-access
```

Comment:
This was something that sqlalchemy v1 vendored for comaptibility with older Python versions: https://github.com/sqlalchemy/sqlalchemy/blob/6cd45888c0f7299125a137d9108c6eff4096561e/lib/sqlalchemy/util/compat.py#L55C10-L55C10
It was removed in v2.0. Since it simply vendored the implementation from Python 3.7, the standard lib is a drop-in replacement